tests/test_metadata_eq_.py

```diff
@@ -0,0 +1,412 @@
+#!/usr/bin/env python
+
+# Copyright New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""Test __eq__ implementations of classes inside tuf/api/metadata.py."""
+
+
+import copy
+import os
+import sys
+import unittest
+from typing import Any, ClassVar, Dict, List
+
+from securesystemslib.signer import SSlibSigner
+
+from tests import utils
+from tests.repository_simulator import RepositorySimulator
+from tuf.api.metadata import (
+    TOP_LEVEL_ROLE_NAMES,
+    DelegatedRole,
+    Delegations,
+    Key,
+    Metadata,
+    MetaFile,
+    Role,
+    TargetFile,
+)
+
+
+# Pylint no-member has to be disabled because of self.case_name.
+# pylint: disable=no-member
+class TestMetadataComparisions(unittest.TestCase):
+    """Test __eq__ for all classes inside tuf/api/metadata.py."""
+
+    metadata: ClassVar[Dict[str, bytes]]
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.repo_dir = os.path.join(
+            utils.TESTS_DIR, "repository_data", "repository", "metadata"
+        )
+        cls.metadata = {}
+        for md in TOP_LEVEL_ROLE_NAMES:
+            with open(os.path.join(cls.repo_dir, f"{md}.json"), "rb") as f:
+                cls.metadata[md] = f.read()
```

Comment:
Also, I think it's shorter if I load the data once in the begging of the class and then reuse it instead of using:
```python
            with open(os.path.join(cls.repo_dir, f"{<ROLE>}.json"), "rb") as f:
                cls.metadata[md] = f.read()
```
in multiple places.

Reply:
Sure, although there is a `Metadata.from_file` method which is just as short. But yeah, it makes sense to read the files only once in setUpClass and I wasn't arguing against that.